web/src/compose_paste.ts

```diff
@@ -40,11 +40,49 @@ function image_to_zulip_markdown(
     return src ? "[" + title + "](" + src + ")" : (node.getAttribute("alt") ?? "");
 }
 
+function count_textful_nodes_in_child_nodes(child_nodes: ChildNode[]): number {
+    let textful_nodes = 0;
+
+    for (const child of child_nodes) {
+        // We do not consider empty childNodes or childNodes containing newlines
+        // while assessing whether the styling on this should be removed.
+        if (child.nodeValue?.trim() === "" || child.nodeName === "#comment") {
+            continue;
+        }
+
+        if (child.nodeValue) {
+            textful_nodes += 1;
+            if (textful_nodes >= 2) {
+                return textful_nodes;
+            }
+        }
+
+        textful_nodes += count_textful_nodes_in_child_nodes([...child.childNodes]);
+
+        if (textful_nodes >= 2) {
+            return textful_nodes;
+        }
+    }
+
+    return textful_nodes;
+}
```

Comment:
This function name seems to be a lie, since it doesn't actually do a count, aborting early once it's found 2 children somewhere? Functions with inaccurate names are usually worse than a function named "foo", at least in our codebase, where we've put a lot of effort into naming functions such that they do what they say.

Reply:
Of course the intention isn't to make the codebase worse -- the point is that **unless** we each give thoughtful attention to the name of every function, we collectively will create that effect through the collective stacking of mistakes.